soh/soh/Enhancements/randomizer/logic.cpp

```diff
@@ -673,6 +673,31 @@ namespace Rando {
             case RE_BIG_OCTO:
                 //If chasing octo is annoying but with rolls you can catch him, and you need rang to get into this room without shenanigains anyway. Bunny makes it free
                 return CanUse(RG_KOKIRI_SWORD) || CanUse(RG_STICKS) || CanUse(RG_MASTER_SWORD);
+            case RE_GOHMA:
+                return HasBossSoul(RG_GOHMA_SOUL) && CanJumpslashExceptHammer() &&
+                    (CanUse(RG_NUTS) || CanUse(RG_FAIRY_SLINGSHOT) || CanUse(RG_FAIRY_BOW) || HookshotOrBoomerang());
+            case RE_KING_DODONGO:
+                return HasBossSoul(RG_KING_DODONGO_SOUL) && CanJumpslashExceptHammer() &&
+                    (CanUse(RG_BOMB_BAG) || HasItem(RG_GORONS_BRACELET));
```

Comment:
Both Gohma and Dodongo can be beaten with hammer so should be using CanJumpslash().